plugin/main/src/kotlinx/benchmark/gradle/JmhBytecodeGeneratorWorker.kt

```diff
@@ -124,8 +133,77 @@ abstract class JmhBytecodeGeneratorWorker : WorkAction<JmhBytecodeGeneratorWorkP
             throw RuntimeException("Generation of JMH bytecode failed with $errCount errors:\n$sb")
         }
     }
+
+    /**
+     * Validates functions annotated with `@Benchmark` and return `true` if no issues were found.
+     * Otherwise, the function returns `false` and logs all detected errors.
+     */
+    private fun validateBenchmarkFunctions(clazz: Class<*>): Boolean {
+        var allFunctionsValid = true
+        // Using declaredMethods to abstain from reporting the same method multiple times in a case
+        // of benchmark classes extending some other classes.
+        clazz.declaredMethods.filter { it.isAnnotationPresent(Benchmark::class.java) }.forEach { method ->
+            allFunctionsValid = allFunctionsValid.and(validateBenchmarkFunction(method))
+        }
+        return allFunctionsValid
+    }
+
+    /**
+     * Validates a benchmark function [function] and return `true` if no issues were found.
+     * Otherwise, the function returns `false` and logs all detected errors.
+     */
+    private fun validateBenchmarkFunction(function: Method): Boolean {
+        val name = function.name
+        if (reservedJavaIdentifierNames.contains(name)) {
+            logger.error(
+                formatInvalidFunctionNameMessage(
+                    "Benchmark function name is a reserved Java keyword and cannot be used", function
+                )
+            )
+            return false
+        }
+
+        if (!Character.isJavaIdentifierStart(name.first()) ||
+            name.substring(1).any { !Character.isJavaIdentifierPart(it) }
+        ) {
+            logger.error(
+                formatInvalidFunctionNameMessage(
+                    "Benchmark function name is not a valid Java identifier", function
+                )
+            )
+            return false
+        }
+
+        return true
+    }
+
+    private fun formatInvalidFunctionNameMessage(errorDescription: String, function: Method): String {
+        val holder = function.declaringClass
+        val javaName = "${holder.canonicalName}.${function.name}"
+        val kotlinName =
+            "${holder.kotlin.qualifiedName ?: holder.name}.${function.kotlinFunction?.name ?: function.name}"
+        return "$errorDescription: \"${javaName}\" (declared as \"$kotlinName\"). " +
+                "This might happen if the function has a backticked (`) name, " +
+                "illegal named specified in @JvmName annotation, or a function returns an inline value class. " +
+                "Consider using @JvmName annotation to provide a valid runtime name."
+    }
 }
 
+private val reservedJavaIdentifierNames = setOf(
+    // boolean and NULL-literals
+    "true", "false", "null",
+    // See Java Language specification, 3.9. Keywords
```

Comment:
Historical note: that section also includes the explanation of why benchmark names like `module`, which, strictly speaking, are an example of a `Keyword` and should be invalid according to the grammar, are actually fine.